src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -199,6 +196,14 @@ AccessibleRepositories getAccessibleRepositories() {
         return getRepositoryAccessStrategy().forContext(context);
     }
 
+    private String actualOwner() {
+        if (getAccessibleRepositories().getOwner() == null) {
+            // Use the possibly inferred owner from the context
+            return context.getInferredOwner();
```

Comment:
Can you add a case to `RunWithCredentialsTest` that covers the new behavior, and that checks the default behavior?